src/mdns.rs

```diff
@@ -20,64 +25,96 @@ const MULTICAST_PORT: u16 = 5353;
 
 /// An mDNS discovery.
 #[allow(non_camel_case_types)]
-pub struct mDNS
-{
+pub struct mDNS {
     /// The name of the service that we are discovering.
     service_name: String,
     /// The UDP sockets that we send/receive multicasts on.
     /// There will be one per socket.
     sockets: Vec<InterfaceDiscovery>,
-    /// The DNS responses we have obtained so far.
-    responses: VecDeque<Response>,
-}
-
-/// An mDNS discovery on a specific interface.
-struct InterfaceDiscovery
-{
-    token: mio::Token,
-    socket: UdpSocket,
+    next_to_poll: usize,
```

Comment:
Recommend renaming this to something like `next_socket_index_to_poll` for clarity.